### ISSUE TITLE
AnalysisException when running Bronze

### ISSUE THREAD

**Issue by j0nnyr0berts** [databrickslabs/overwatch#140]
Hi,

When running `Bronze.workspace(run)` I get the following exception:

```
AnalysisException: cannot resolve '`state`' given input columns: [spark_catalog.overwatch_etl.clusters_snapshot_bronze.Overwatch_RunID, spark_catalog.overwatch_etl.clusters_snapshot_bronze.Pipeline_SnapTS, spark_catalog.overwatch_etl.clusters_snapshot_bronze.aws_attributes, spark_catalog.overwatch_etl.clusters_snapshot_bronze.custom_tags, spark_catalog.overwatch_etl.clusters_snapshot_bronze.organization_id, snapRnk, spark_catalog.overwatch_etl.clusters_snapshot_bronze.spark_conf, spark_catalog.overwatch_etl.clusters_snapshot_bronze.spark_env_vars, spark_catalog.overwatch_etl.clusters_snapshot_bronze.value];
```

Full trace:

```
'Filter ('state = RUNNING)
+- Filter (snapRnk#5754 = 1)
   +- Window [value#5736, organization_id#5737, custom_tags#5738, spark_conf#5739, spark_env_vars#5740, aws_attributes#5741, Pipeline_SnapTS#5742, Overwatch_RunID#5743, rank(Pipeline_SnapTS#5742) windowspecdefinition(organization_id#5737, Pipeline_SnapTS#5742 DESC NULLS LAST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS snapRnk#5754], [organization_id#5737], [Pipeline_SnapTS#5742 DESC NULLS LAST]
      +- Project [value#5736, organization_id#5737, custom_tags#5738, spark_conf#5739, spark_env_vars#5740, aws_attributes#5741, Pipeline_SnapTS#5742, Overwatch_RunID#5743]
         +- Filter (organization_id#5737 = 337210421809158)
            +- SubqueryAlias spark_catalog.overwatch_etl.clusters_snapshot_bronze
               +- Relation[value#5736,organization_id#5737,custom_tags#5738,spark_conf#5739,spark_env_vars#5740,aws_attributes#5741,Pipeline_SnapTS#5742,Overwatch_RunID#5743] parquet

	at org.apache.spark.sql.catalyst.analysis.package$AnalysisErrorAt.failAnalysis(package.scala:42)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$$anonfun$$nestedInanonfun$checkAnalysis$2$2.applyOrElse(CheckAnalysis.scala:174)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$$anonfun$$nestedInanonfun$checkAnalysis$2$2.applyOrElse(CheckAnalysis.scala:165)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformUp$2(TreeNode.scala:349)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:81)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformUp(TreeNode.scala:349)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformUp$1(TreeNode.scala:346)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$mapChildren$1(TreeNode.scala:415)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:251)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:413)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:366)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformUp(TreeNode.scala:346)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.$anonfun$transformExpressionsUp$1(QueryPlan.scala:105)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.$anonfun$mapExpressions$1(QueryPlan.scala:117)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:81)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.transformExpression$1(QueryPlan.scala:117)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.recursiveTransform$1(QueryPlan.scala:128)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.$anonfun$mapExpressions$4(QueryPlan.scala:138)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:251)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.mapExpressions(QueryPlan.scala:138)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.transformExpressionsUp(QueryPlan.scala:105)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.$anonfun$checkAnalysis$2(CheckAnalysis.scala:165)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.$anonfun$checkAnalysis$2$adapted(CheckAnalysis.scala:98)
	at org.apache.spark.sql.catalyst.trees.TreeNode.foreachUp(TreeNode.scala:191)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.$anonfun$checkAnalysis$1(CheckAnalysis.scala:98)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.databricks.spark.util.FrameProfiler$.record(FrameProfiler.scala:80)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.checkAnalysis(CheckAnalysis.scala:95)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.checkAnalysis$(CheckAnalysis.scala:95)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.checkAnalysis(Analyzer.scala:204)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$executeAndCheck$1(Analyzer.scala:227)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.markInAnalyzer(AnalysisHelper.scala:232)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.executeAndCheck(Analyzer.scala:224)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$analyzed$1(QueryExecution.scala:96)
	at com.databricks.spark.util.FrameProfiler$.record(FrameProfiler.scala:80)
	at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:134)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:178)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:852)
	at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:178)
	at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:97)
	at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:94)
	at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:86)
	at org.apache.spark.sql.Dataset.<init>(Dataset.scala:215)
	at org.apache.spark.sql.Dataset.<init>(Dataset.scala:221)
	at org.apache.spark.sql.Dataset$.apply(Dataset.scala:79)
	at org.apache.spark.sql.Dataset.withTypedPlan(Dataset.scala:3828)
	at org.apache.spark.sql.Dataset.filter(Dataset.scala:1610)
	at com.databricks.labs.overwatch.pipeline.TransformFunctions$.getClusterIdsWithNewEvents(TransformFunctions.scala:422)
	at com.databricks.labs.overwatch.pipeline.BronzeTransforms.prepClusterEventLogs(BronzeTransforms.scala:314)
	at com.databricks.labs.overwatch.pipeline.BronzeTransforms.prepClusterEventLogs$(BronzeTransforms.scala:303)
	at com.databricks.labs.overwatch.pipeline.Bronze.prepClusterEventLogs(Bronze.scala:8)
	at com.databricks.labs.overwatch.pipeline.Bronze.appendClusterEventLogsProcess$lzycompute(Bronze.scala:75)
	at com.databricks.labs.overwatch.pipeline.Bronze.appendClusterEventLogsProcess(Bronze.scala:68)
	at com.databricks.labs.overwatch.pipeline.Bronze.$anonfun$executeModules$1(Bronze.scala:128)
	at com.databricks.labs.overwatch.pipeline.Bronze.$anonfun$executeModules$1$adapted(Bronze.scala:125)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at com.databricks.labs.overwatch.pipeline.Bronze.executeModules(Bronze.scala:125)
	at com.databricks.labs.overwatch.pipeline.Bronze.run(Bronze.scala:146)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(command-1029368660618364:1)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(command-1029368660618364:53)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read$$iw$$iw$$iw$$iw$$iw$$iw.<init>(command-1029368660618364:55)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read$$iw$$iw$$iw$$iw$$iw.<init>(command-1029368660618364:57)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read$$iw$$iw$$iw$$iw.<init>(command-1029368660618364:59)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read$$iw$$iw$$iw.<init>(command-1029368660618364:61)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read$$iw$$iw.<init>(command-1029368660618364:63)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read$$iw.<init>(command-1029368660618364:65)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read.<init>(command-1029368660618364:67)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read$.<init>(command-1029368660618364:71)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$read$.<clinit>(command-1029368660618364)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$eval$.$print$lzycompute(<notebook>:7)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$eval$.$print(<notebook>:6)
	at $lineb7468750a5ed4474961a9b981330dd9c41.$eval.$print(<notebook>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.call(IMain.scala:745)
	at scala.tools.nsc.interpreter.IMain$Request.loadAndRun(IMain.scala:1021)
	at scala.tools.nsc.interpreter.IMain.$anonfun$interpret$1(IMain.scala:574)
	at scala.reflect.internal.util.ScalaClassLoader.asContext(ScalaClassLoader.scala:41)
	at scala.reflect.internal.util.ScalaClassLoader.asContext$(ScalaClassLoader.scala:37)
	at scala.reflect.internal.util.AbstractFileClassLoader.asContext(AbstractFileClassLoader.scala:41)
	at scala.tools.nsc.interpreter.IMain.loadAndRunReq$1(IMain.scala:573)
	at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:600)
	at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:570)
	at com.databricks.backend.daemon.driver.DriverILoop.execute(DriverILoop.scala:219)
	at com.databricks.backend.daemon.driver.ScalaDriverLocal.$anonfun$repl$1(ScalaDriverLocal.scala:235)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.databricks.backend.daemon.driver.DriverLocal$TrapExitInternal$.trapExit(DriverLocal.scala:852)
	at com.databricks.backend.daemon.driver.DriverLocal$TrapExit$.apply(DriverLocal.scala:805)
	at com.databricks.backend.daemon.driver.ScalaDriverLocal.repl(ScalaDriverLocal.scala:235)
	at com.databricks.backend.daemon.driver.DriverLocal.$anonfun$execute$12(DriverLocal.scala:494)
	at com.databricks.logging.UsageLogging.$anonfun$withAttributionContext$1(UsageLogging.scala:240)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at com.databricks.logging.UsageLogging.withAttributionContext(UsageLogging.scala:235)
	at com.databricks.logging.UsageLogging.withAttributionContext$(UsageLogging.scala:232)
	at com.databricks.backend.daemon.driver.DriverLocal.withAttributionContext(DriverLocal.scala:50)
	at com.databricks.logging.UsageLogging.withAttributionTags(UsageLogging.scala:279)
	at com.databricks.logging.UsageLogging.withAttributionTags$(UsageLogging.scala:271)
	at com.databricks.backend.daemon.driver.DriverLocal.withAttributionTags(DriverLocal.scala:50)
	at com.databricks.backend.daemon.driver.DriverLocal.execute(DriverLocal.scala:471)
	at com.databricks.backend.daemon.driver.DriverWrapper.$anonfun$tryExecutingCommand$1(DriverWrapper.scala:688)
	at scala.util.Try$.apply(Try.scala:213)
	at com.databricks.backend.daemon.driver.DriverWrapper.tryExecutingCommand(DriverWrapper.scala:680)
	at com.databricks.backend.daemon.driver.DriverWrapper.getCommandOutputAndError(DriverWrapper.scala:521)
	at com.databricks.backend.daemon.driver.DriverWrapper.executeCommand(DriverWrapper.scala:633)
	at com.databricks.backend.daemon.driver.DriverWrapper.runInnerLoop(DriverWrapper.scala:426)
	at com.databricks.backend.daemon.driver.DriverWrapper.runInner(DriverWrapper.scala:369)
	at com.databricks.backend.daemon.driver.DriverWrapper.run(DriverWrapper.scala:221)
	at java.lang.Thread.run(Thread.java:748)
```

Output:

```
+--------+----------------------------+--------------------+-------------------+-------------------+-------------------+
|moduleID|moduleName                  |primordialDateString|fromTS             |untilTS            |snapTS             |
+--------+----------------------------+--------------------+-------------------+-------------------+-------------------+
|1002    |Bronze_Clusters_Snapshot    |2021-04-01          |2021-06-18 09:28:28|2021-06-21 08:59:40|2021-06-21 08:59:40|
|1003    |Bronze_Pools                |2021-04-01          |2021-06-18 09:28:28|2021-06-21 08:59:40|2021-06-21 08:59:40|
|1004    |Bronze_AuditLogs            |2021-04-01          |2021-06-18 09:28:28|2021-06-21 08:59:40|2021-06-21 08:59:40|
|2003    |Silver_SPARK_Executors      |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|2005    |Silver_SPARK_Executions     |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|2006    |Silver_SPARK_Jobs           |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|2007    |Silver_SPARK_Stages         |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|2008    |Silver_SPARK_Tasks          |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|2010    |Silver_JobsStatus           |2021-04-01          |2021-06-18 09:28:28|2021-06-21 08:59:40|2021-06-21 08:59:40|
|2011    |Silver_JobsRuns             |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|2018    |Silver_Notebooks            |2021-04-01          |2021-06-18 09:28:28|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3001    |Gold_Cluster                |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3002    |Gold_Job                    |2021-04-01          |2021-06-18 09:28:28|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3003    |Gold_JobRun                 |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3004    |Gold_Notebook               |2021-04-01          |2021-06-18 09:28:28|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3005    |Gold_ClusterStateFact       |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3010    |Gold_SparkJob               |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3011    |Gold_SparkStage             |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3012    |Gold_SparkTask              |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3013    |Gold_SparkExecution         |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3014    |Gold_SparkExecutor          |2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
|3015    |Gold_jobRunCostPotentialFact|2021-04-01          |2021-04-01 00:00:00|2021-06-21 08:59:40|2021-06-21 08:59:40|
+--------+----------------------------+--------------------+-------------------+-------------------+-------------------+

Overriding spark.databricks.delta.optimize.maxFileSize from 1073741824 --> 134217728
Overriding spark.databricks.delta.optimizeWrite.numShuffleBlocks from 50000000 --> 50000
Overriding spark.databricks.delta.optimizeWrite.binSize from 536870912b --> 512
DEBUG: CLOUD PROVIDER = aws
Beginning: Bronze_AuditLogs
MODULE: 1004-Bronze_AuditLogs
TIME RANGE:     From -> To == 2021-06-18 09:28:28 -> 2021-06-21 08:59:40
Bronze_AuditLogs: Validating Input Schemas
SUCCESS! Bronze_AuditLogs: 28770 records appended.
Beginning: Bronze_Pools
MODULE: 1003-Bronze_Pools
TIME RANGE:     From -> To == 2021-06-18 09:28:28 -> 2021-06-21 08:59:40
Bronze_Pools: Validating Input Schemas
SUCCESS! Bronze_Pools: 1 records appended.
Beginning: Bronze_Clusters_Snapshot
MODULE: 1002-Bronze_Clusters_Snapshot
TIME RANGE:     From -> To == 2021-06-18 09:28:28 -> 2021-06-21 08:59:40
Bronze_Clusters_Snapshot: Validating Input Schemas
SUCCESS! Bronze_Clusters_Snapshot: 1 records appended.
```

Param string:

```
res4: String = {"auditLogConfig":{"rawAuditPath":"dbfs:/mnt/audit-logs/workspaceId=337210421809158/"},"tokenSecret":{"scope":"overwatch-scope","key":"api-token"},"dataTarget":{"databaseName":"overwatch_etl","databaseLocation":"dbfs:/mnt/overwatch/337210421809158/overwatch_etl.db","etlDataPathPrefix":"dbfs:/mnt/overwatch/global_share","consumerDatabaseName":"overwatch","consumerDatabaseLocation":"dbfs:/mnt/overwatch/337210421809158/overwatch.db"},"badRecordsPath":"dbfs:/mnt/audit-logs/337210421809158/sparkEventsBadrecords","overwatchScope":["audit","sparkEvents","jobs","clusters","clusterEvents","notebooks","pools"],"maxDaysToLoad":90,"databricksContractPrices":{"interactiveDBUCostUSD":0.56,"automatedDBUCostUSD":0.26},"primordialDateString":"2021-04-01"}
```

Both Silver and Gold jobs complete fine. Any help appreciated 🙂

**Comment by GeekSheikh** [comment]
I will be able to look into this later today. In the meantime, please send me your pipeline report table. It'll be the output of the `display(pipReport(etlDB))` command at the very bottom of the example notebook.

**Comment by j0nnyr0berts** [comment]
Thank you!

[export.csv](https://github.com/databrickslabs/overwatch/files/6686423/export.csv)

**Comment by GeekSheikh** [comment]
From your export.csv I can see that you are only loading data for one cluster. If this is a brand new workspace, it's possible to have only one cluster but I suspect that you have more than that but whichever credentials you're using don't have appropriate access to read the cluster details. The `state` column is a part of the schema for the api call `clusters/list` which is stored in Module 1002 in table `clusters_snapshot_bronze`. As you can see from the [api response structure](https://docs.databricks.com/dev-tools/api/latest/clusters.html#clusterinfo), the state is a field that should be returned. If you do not have appropriate access to any clusters to return this data, Overwatch will not run. 

Please review [Configuration-Security](https://databrickslabs.github.io/overwatch/gettingstarted/configuration/#tokensecret) section of the docs. The account under which Overwatch executes must have appropriate read access to the assets that are expected to be ingested.

I hope this helps. If you feel that you do have the appropriate access, please verify by logging in with the account Overwatch is executing under and go to the clusters tab --> `accessible by me` --> select a cluster --> `Event Log`. If the Overwatch user cannot access any Cluster Event Logs, you cannot proceed with the clusterEvents scope. Please note that all of your api modules are returning only 1 record. It appears you don't have the necessary access.

**Comment by j0nnyr0berts** [comment]
Thank you for your detailed response. You're completely correct - while the user had the necessary perms, the API key had expired. I am now able to run all the stages!

I'm getting the below. I'm wondering if it's because my cluster logs are stored in S3 and not in the dbfs mount or because I only have cluster logs setup for one cluster. I have tried deleting all Overwatch files and reloading but no change.

```
EMPTY: 1006-Bronze_SparkEventLogs Module: SKIPPING Downstream modules that depend on this module will not progress until new data is received by this module. Module Dependencies: 1004 Event Logs DF is empty, Exiting
```

I have many other EMPTY statuses but I think they are related to the above. However, I'm also seeing the below error which seems unrelated (but perhaps isn't!).  Let me know if you'd like me to create a separate a new issue. Thanks again.


```
FAILED --> : ERROR: Silver_JobsRuns FAILED --> Message: cannot resolve '`clusterId`' given input columns: [jobId, spark_catalog.overwatch_etl.jobs_snapshot_bronze.organization_id, timestamp]; 'Filter isnotnull('clusterId) +- Project [organization_id#128576, job_id#128574L AS jobId#128586L, created_time#128572L AS timestamp#128587L] +- Filter (organization_id#128576 = 337210421809158) +- SubqueryAlias spark_catalog.overwatch_etl.jobs_snapshot_bronze +- Relation[created_time#128572L,creator_user_name#128573,job_id#128574L,settings#128575,organization_id#128576,Pipeline_SnapTS#128577,Overwatch_RunID#128578] parquet
```

**Comment by GeekSheikh** [comment]
Glad you're able to run the stages now.

The first error has nothing to do with S3 or cluster logs, this is referring to the event logs of the cluster. I'm not sure what your pipeline state is in at this point but it's possible that you need to drop everything and start over with the appropriate access. Either the user does not have access to the cluster events or there simply aren't any within the time frame that's being executed on, this can be seen in the pipReport in the from/to columns.

The second error seems to be a missing column. I've never hit this issue before...but...I've also never tried to run Overwatch jobs module when there weren't any jobs that the user either didn't have access to or that simply didn't exist. Overwatch isn't meant to be run in a playground and when it is, there are often unexpected columns missing that are required for the ETL. A jobRun requires a clusterId, if there are various components missing it will cause issues. This issue is saying there isn't any real, relevant data in the jobs_snapshot_bronze table which goes back to the original issue of no access. If you now have access to read the jobs, it's possible that the Overwatch state moved on ahead and none of the data was loaded. You'll either need to rollback the appropriate tables or more simplistically in this case, just drop everything and re-run when appropriate access is restored.

**Comment by j0nnyr0berts** [comment]
Got it, thanks! I did give dropping all and rerunning with the permissions restored a go but ran into the same issue - I will keep playing. I've got the cluster module coming through and I'm mainly interested in the costs data so I am in a good place. Thanks again for all your help!

**Comment by GeekSheikh** [comment]
@j0nnyr0berts, just took a glance at the code, the second error could be because you've never executed a job on an existing cluster and/or the state (described in last comment) and/or because the Overwatch user doesn't have access to any jobs that have executed on an existing cluster. Could you please review and confirm which of these you're hitting?

**Comment by j0nnyr0berts** [comment]
The profile I am using for Overwatch (my own) has full admin access - I can read the event logs of every cluster in the workspace (they are all populated). 

I've deleted all the Overwatch files, dropped the overwatch and overwatch_etl databases and run overwatch again but getting same output I'm afraid. 

Apologies if these steps are not adequately testing the conditions you mentioned - let me know if there's anything else I can do!

**Comment by GeekSheikh** [comment]
Thanks for the follow-up info.

Do you have any jobs in your workspace that have executed on an interactive (i.e. existing) cluster? If not, please create a little test job that runs on an existing interactive cluster. This job can simply point to a notebook that adds 1+1. Once the job is complete, please re-run Overwatch. If this fixes the issue, please note that this is fixed in the next version but this is a work-around for now.

**Comment by j0nnyr0berts** [comment]
Created a simple job and ran on an interactive cluster - no change I'm afraid. Happy to try again later if you think there could be some lag in the logs being created and delivered to s3.

**Comment by GeekSheikh** [comment]
correct, audit logs are only delivered daily. So you will need to run the test in the previous comment 1 or more days after you execute that simple job, sorry for not including that detail above.

**Comment by j0nnyr0berts** [comment]
Ah great, thanks - will give it another run when I start tomorrow and let you know how I get on.

**Comment by j0nnyr0berts** [comment]
Worked perfectly. Thank you for your patience!
